micrometer-core/src/main/java/io/micrometer/core/instrument/binder/grpc/GrpcClientObservationContext.java

```diff
@@ -32,110 +32,77 @@
  */
 public class GrpcClientObservationContext extends RequestReplySenderContext<Metadata, Object> {
 
-    private String serviceName;
-
-    private String methodName;
-
-    private String fullMethodName;
-
-    private MethodType methodType;
-
-    @Nullable
-    private Code statusCode;
-
-    private String authority;
-
-    private Metadata headers;
-
-    private Metadata trailers;
+    private GrpcObservationContextInfo grpcInfo;
 
     public GrpcClientObservationContext(Setter<Metadata> setter) {
         super(setter);
+        this.grpcInfo = new GrpcObservationContextInfo();
     }
 
+    // Delegates to grpcInfo
     public String getServiceName() {
-        return this.serviceName;
+        return grpcInfo.getServiceName();
     }
 
-    public void setServiceName(String serviceName) {
-        this.serviceName = serviceName;
+    public String getMethodName() {
```

Comment:
Reordering the methods makes this PR hard to review and also, it is not what the PR claims as its goal. If you want to increase readability by reordering the methods, I think that should be a separate PR or it should not do such thing.

Also, removing javadoc is something that the automation should not do, it is removing valuable information.